compiler/backend/ccgstmts.nim

```diff
@@ -729,28 +729,41 @@ proc genAsmOrEmitStmt(p: BProc, t: PNode, isAsmStmt=false): Rope =
       res.add(it.strVal)
     of nkSym:
       var sym = it.sym
-      if sym.kind in {skProc, skFunc, skIterator, skMethod}:
+      case sym.kind
+      of skProc, skFunc, skIterator, skMethod:
         var a: TLoc
         initLocExpr(p, it, a)
-        res.add($rdLoc(a))
-      elif sym.kind == skType:
-        res.add($getTypeDesc(p.module, sym.typ))
-      else:
+        res.add(rdLoc(a))
+      of skVar, skLet, skForVar, skParam, skResult, skTemp:
+        # make sure the C type description is available:
+        discard getTypeDesc(p.module, skipTypes(sym.typ, abstractPtrs))
+        var a: TLoc
+        initLocExpr(p, it, a)
+        res.add(rdLoc(a))
+      of skType:
+        res.add(getTypeDesc(p.module, sym.typ))
+      of skField:
+        # special support for raw field symbols
         discard getTypeDesc(p.module, skipTypes(sym.typ, abstractPtrs))
         var r = sym.loc.r
         if r == "":
-          # if no name has already been given,
-          # it doesn't matter much:
-          r = mangleName(p.module, sym)
+          # problem: a field doesn't know what type it is part of, so we
```

Comment:
Would `sym.owner.typ` not provide the owning type or perhaps raw field symbols are different?

Reply:
Unfortunately, no, the `owner` of field symbols cannot be used here. This is because:
- for instances of generic objects, the field owner points to the symbol of the generic object, and from there it's not possible to get the instantiated type
- in some circumstances, the owner for object fields doesn't point to an `skType` symbol
- fields of named tuples cannot point to the tuple type's symbol (because there is none)

Reply:
sem lets us down yet again. :D

OK, let's merge this and I'll see if I can maybe plug a few of these holes.